pkg/extensions/tbtc/recovery/storage_test.go

```diff
@@ -0,0 +1,57 @@
+package recovery
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestDerivationIndexStorage_Read(t *testing.T) {
+	dir, err := ioutil.TempDir("", "example")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	dis, err := NewDerivationIndexStorage(dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	index, err := dis.Read("extendedPublicKey")
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedIndex := 0
+	if index != expectedIndex {
+		t.Errorf("incorrect extendedPublicKey index\nexpected: %d\nactual:   %d", expectedIndex, index)
```

Comment:
Should we be able to check if there is something in the registry for the particular public key? This is in case it is used for the first time or if it was altered in the config.

Reply:
can you elaborate here?

Reply:
I was not sure how the storage code will be used and I was thinking whether there should be a function allowing to query the storage to see if the public key with the given index was used or not. It seems that such functions are not necessary though.